src/helpers/__tests__/utils.test.ts

```diff
@@ -211,4 +215,224 @@ describe('utils', () => {
       expect(ciUtils.getProxyUrl({auth, protocol: 'http'})).toBe('')
     })
   })
+
+  // Test the different possibilities of proxy configuration of getRequestHelper.
+  // All the calls to getRequestHelpers should be https calls, but to keep the test suite
+  // simple tests are using http calls (testing with https would require us to add http certs
+  // and configure axios to trust these https certs, which requires an agent config, which
+  // interferes a bit with how the proxies are configured since they are configured through an
+  // agent themselves.
+  // Proxy of https requests is still tested in the proxy-agent library itself.
+  describe('Proxy configuration', () => {
+    const initialHttpProxyEnv = process.env.HTTP_PROXY
```

Comment:
Really nit: It does not really change the behavior but for symmetry you could put this in a `beforeAll` block